src/algebra/group/pi.lean

```diff
@@ -133,38 +133,27 @@ into a dependent family of values, as functions supported at a point.
 This is the `zero_hom` version of `pi.single`. -/
 @[simps] def zero_hom.single [Π i, has_zero $ f i] (i : I) : zero_hom (f i) (Π i, f i) :=
 { to_fun := single i,
-  map_zero' := function.update_eq_self i 0 }
+  map_zero' := single_zero i }
 
 /-- The additive monoid homomorphism including a single additive monoid
 into a dependent family of additive monoids, as functions supported at a point.
 
 This is the `add_monoid_hom` version of `pi.single`. -/
 @[simps] def add_monoid_hom.single [Π i, add_monoid $ f i] (i : I) : f i →+ Π i, f i :=
 { to_fun := single i,
-  map_add' := λ x y, funext $ λ j, begin
-    refine (apply_single₂ _ (λ _, _) i x y j).symm,
-    exact zero_add 0,
-  end,
+  map_add' := single_op₂ (λ _, (+)) (λ _, zero_add _) _,
   .. (zero_hom.single f i) }
 
 /-- The multiplicative homomorphism including a single `monoid_with_zero`
```

Comment:
The docstring does not correspond to the definition any more.

Reply:
```suggestion
/-- The multiplicative homomorphism including a single `mul_zero_class`
```